pkg/async-gateway/service.go

```diff
@@ -52,25 +55,42 @@ func NewService(clusterUID, apiName string, queue Queue, storage Storage, logger
 }
 
 // CreateWorkload enqueues an async workload request and uploads the request payload to S3
-func (s *service) CreateWorkload(id string, payload io.Reader, contentType string) (string, error) {
+func (s *service) CreateWorkload(id string, payload io.Reader, headers http.Header) (string, error) {
+	contentType := headers.Get("Content-Type")
+	if contentType == "" {
+		return "", errors.ErrorUnexpected("missing content-type in headers")
+	}
+	headers.Del("Content-Type")
```

Comment:
Since we pass all the headers through now, should we consider preserving the "Content-Type" header here so it gets uploaded with all the others, and then not setting it explicitly in the dequeuer (i.e. deleting line 189 in `pkg/dequeuer/async_handler.go`)? Theoretically we might not even need to set the content type on the s3 object, but I think that is still nice, and there's no reason not to do it.

Reply:
Also, perhaps now we could allow the user to not specify the content type header? We can discuss this on Monday

Reply:
I agree with this. I think making sure that the `Content-Type` is present in the headers and removing it from the `userPayload` struct so that the payload is only meant for the _payload_ makes sense to me.

---

Not asking the users to specify the content type can lead to a less opinionated implementation. Also, from [RFC 7231, section 3.1.1.5](https://datatracker.ietf.org/doc/html/rfc7231#section-3.1.1.5)
```
   A sender that generates a message containing a payload body SHOULD
   generate a Content-Type header field in that message unless the
   intended media type of the enclosed representation is unknown to the
   sender.  If a Content-Type header field is not present, the recipient
   MAY either assume a media type of "application/octet-stream"
   ([RFC2046], Section 4.5.1) or examine the data to determine its type.
```
What that says is that if the sender doesn't know what the content type is, then it's okay if it doesn't add it. So we shouldn't enforce it.